### PR TITLE
Added Highlight inherited permissions

### PR DESCRIPTION
**Pull request by minsunny**
https://github.com/foam-framework/foam2/issues/1352#event-1748292327

1. changed parent type string to reference in Group model
2. added highlights inherited permission and made it updated along parent

### CODE REVIEW

File: src/foam/nanos/auth/PermissionTableView.js [hunk 1]
```diff
@@ -94,7 +97,17 @@ foam.CLASS({
                               cb.data$.sub(function() { self.updateGroup(p, g, cb.data); });
                               this.start('td').show(groups[g.id].data$.map(function() {return groups[g.id].data;}))
                                     .style({'text-align': 'center', 'width': '100'}).tag(cb).call(function() {
-                                      if ( g.implies(p.id) ) { cb.style({'border-color': '#40C75B'}) };
+                                      self.groupDAO.find(g.parent).then(function(a){
```
Comment (kgrgreer) [inline src/foam/nanos/auth/PermissionTableView.js]: Space between ) and {.

Comment (kgrgreer) [inline src/foam/nanos/auth/PermissionTableView.js]: You can do g.parent$find(function(a) { , which is shorter.

File: src/foam/nanos/auth/PermissionTableView.js [hunk 2]
```diff
@@ -94,7 +97,17 @@ foam.CLASS({
...
+                                        if ( a != undefined && a.implies(p.id) ) {
+                                          cb.setAttribute('title', g.parent + ': ' + p.id); //'Permission inherited from ' + g.parent
+                                          cb.style({'border-color': '#40C75B'})
+                                        }
+                                      });
+
+                                      if ( g.implies(p.id) ) {
+                                        cb.setAttribute('title', g.id + ': ' + p.id);  //'Permissions granted from ' + g.id
+                                        cb.style({'border-color': '#40C75B'});
+                                      }
```
Comment (kgrgreer) [inline src/foam/nanos/auth/PermissionTableView.js]: Move the parent group code above into the else block of this 'if' to avoid extra work and avoid a race condition.

File: src/foam/nanos/auth/PermissionTableView.js [hunk 3]
```diff
@@ -116,17 +129,33 @@ foam.CLASS({
...
+      var e = foam.mlang.Expressions.create();
+
       dao.find(group).then(function(group) {
         // Remove permission if found
         var permissions = group.permissions.filter(function(p) {
           return p.id != permission.id;
         });
 
+        //parents' permissions
+        dao.find(group.parent).then(function(groupParent) {
```
Comment (kgrgreer) [inline src/foam/nanos/auth/PermissionTableView.js]: group.parent$find(function(groupParent) {

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 4]
```diff
@@ -43,7 +43,7 @@ public static EmailTemplate findTemplate(X x, String name, String groupId) {
       }
 
       Group group = (Group) groupDAO.find(groupId);
-      groupId = ( group != null && ! SafetyUtil.isEmpty(group.getParent()) ) ? group.getParent() : "*";
+      groupId = ( group != null && ! SafetyUtil.isEmpty(group.getParent().toString()) ) ? group.getParent().toString() : "*";
```
Comment (kgrgreer) [inline src/foam/nanos/notification/email/DAOResourceLoader.java]: Why is this change needed?

File: src/foam/nanos/auth/PermissionTableView.js [hunk 5]
```diff
@@ -92,9 +95,19 @@ foam.CLASS({
...
                                     .style({'text-align': 'center', 'width': '100'}).tag(cb).call(function() {
-                                      if ( g.implies(p.id) ) { cb.style({'border-color': '#40C75B'}) };
+                                      self.groupDAO.find(g.parent).then(function(a) {
+                                        if ( a != undefined && a.implies(p.id) ) {
+                                          cb.setAttribute('title', g.parent + ': ' + p.id);
+                                          cb.style({'border-color': '#40C75B'})
+                                        }
+                                      });
+
+                                      if ( g.implies(p.id) ) {
```
Comment (kgrgreer) [inline src/foam/nanos/auth/PermissionTableView.js]: If g.implies(p.id) is true then you don't need to do the self.groupDAO.find() above, so move that code into an 'else' block of this if statement.

File: src/foam/nanos/notification/email/DAOResourceLoader.java (hunk 4 above)
Comment (minsunny) [reply src/foam/nanos/notification/email/DAOResourceLoader.java]: getParent() already returns a String so the toString() isn't needed.
